FWCore/Framework/interface/InputSource.h

```diff
@@ -414,7 +417,7 @@ namespace edm {
     virtual void endRun(Run&);
     virtual void beginJob();
     virtual void endJob();
-    virtual SharedResourcesAcquirer* resourceSharedWithDelayedReader_() const;
+    virtual SharedResourcesAcquirer* resourceSharedWithDelayedReader_();
 
```

Comment:
OK to return ptr to non-const in const function because the SharedResourcesAcquirer is mutable.